scipy/stats/stats.py

```diff
@@ -5718,3 +5720,179 @@ def rankdata(a, method='average'):
 
     # average method
     return .5 * (count[dense] + count[dense - 1] + 1)
+
+
+def rvs_ratio_unif(f, size=1, x1=None, x2=None, y2=None, shift=0, maxiter=None,
+                   bounds=None):
+    """
+    Generate random samples from a probability density function using the
+    ratio-of-uniforms method.
+
+    Given a univariate probability density function (pdf) f and a constant c,
+    define the set A(c) = {(x, y) : 0 < y <= sqrt(f(x/y + c)) }.
+    If (U, V) is a random vector uniformly distributed over A(c), then U/V + c
+    follows a distribution with pdf f.
+
+    The above result (see [1]_, [2]_) can be used to sample random variables
+    using only the pdf, i.e. no inversion of the cdf is required. Typical
+    choices of c are zero or the mode of f. The set A(c) is a subset of the
+    rectangle R = [-x1, x2] x [0, y2] where
+    - x1 = inf (x - c) sqrt(f(x))
+    - x2 = sup (x - c) sqrt(f(x))
+    - y2 = sup sqrt(f(x))
+
+    In particular, these values are finite if f is bounded and x**2 * f(x) is
+    bounded (i.e. subquadratic tails). One can generate (U, V) uniformly on
+    R and return U/V + c if (U, V) are also in A(c) which can be directly
+    verified.
+
+    Parameters
+    ----------
+    f : callable
+        The pdf of the distribution.
+    size : int, optional
+        The number of random variables to generate.
+    x1 : float, optional
+        The lower bound of the bounding rectangle in the x-direction.
+        Default is None. In that case, attempt to find bound numerically.
+    x2 : float, optional
+        The upper bound of the bounding rectangle in the x-direction.
+        Default is None. In that case, attempt to find bound numerically.
+    y2 : float, optional
+        The upper bound of the bounding rectangle in the y-direction.
+        Default is None. In that case, attempt to find bound numerically.
+    maxiter : int, optional
+        Maximum number of iterations to generate the random variables.
+        Default is None. In that case, maxiter is set to a value that
+        ensures successful simulation with a high probability.
+    bounds : tuple of floats (a, b), optional
+        Default is None. In case any of the values x1, x2, y2 are not
+        specified (i.e. equal to None), attempt to find the value numerically
+        over the interval from a to b.
+
+    Returns
+    -------
+    rvs : array
+        array with the random variables distributed according to pdf f
```

Comment:
the random variables distributed according to the probability distribution, `pdf`